### DISCUSSION TITLE
How to install ApostropheCMS in GitHub Codespaces + Atlas MongoDB

### DISCUSSION THREAD

**Discussion by ezzle** [apostrophecms/apostrophe#3999] [Q&A] [answered]
Hi.

How to install successfully on a codespaces environment with a cloud-based atlas MongoDB database?

`MongoServerSelectionError: connect ECONNREFUSED 127.0.0.1:27017
    at Timeout._onTimeout (/workspaces/apos-esse/apos-esse/node_modules/mongodb/lib/core/sdam/topology.js:438:30)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7) {
  reason: TopologyDescription {
    type: 'Single',
    setName: null,
    maxSetVersion: null,
    maxElectionId: null,
    servers: Map(1) { 'localhost:27017' => [ServerDescription] },
    stale: false,
    compatible: true,
    compatibilityError: null,
    logicalSessionTimeoutMinutes: null,
    heartbeatFrequencyMS: 10000,
    localThresholdMS: 15,
    commonWireVersion: null
  }
}`

Best regards

**Comment by boutell** [comment, marked as answer]
Hi Ezzle,

Set the `APOS_MONGODB_URI` environment variable to the appropriate MongoDB URI, or set the `uri` option to the `@apostrophecms/db` module.

**Reply by ezzle** [reply]
Thank you!
Worked fine with option
